### PR TITLE
Classic algorithms implementation

### PR DESCRIPTION
**Pull request by dh4618**
- implement potential field algorithm in n dimensions
- implement wave-front algorithm in n dimensions
- add large 3d maps for testing - works for all graph-based algorithms except wave-front (takes too long)
- implement all rrt algorithms in n dimensions - test passes with large 3d examples

### CODE REVIEW

File: src/algorithms/classic/graph_based/potential_field.py [hunk 1]
```diff
@@ -97,30 +94,22 @@ def set_display_info(self) -> List[MapDisplay]:
...
-                    uo = self.calc_repulsive_potential(x=x, y=y, ox=([x.position.x for x in grid.obstacles]), 
-                    oy=([y.position.y for y in grid.obstacles]), rr=(grid.agent.radius))
-                else:
-                    uo = 0
+        pmap = np.zeros(grid.size, dtype=np.float32)
+        #print(str(grid.size))
+
+        for index in np.ndindex(*grid.size):
+            ug = self.calc_attractive_potential(index, grid.goal.position)
+            if Point(*index) not in grid.obstacles:
```
Comment (Nabushika) [inline src/algorithms/classic/graph_based/potential_field.py:104]: is `grid.obstacles` always populated? Maybe best to use `is_agent_valid_pos`, no?

File: src/algorithms/classic/graph_based/potential_field.py [hunk 2]
```diff
@@ -147,17 +136,19 @@ def calc_potential_field(self,grid: Map):
...
 
-    def calc_attractive_potential(self,x, y, gx, gy):
-        return 0.5 * self.KP * np.hypot(x - gx, y - gy)
+    def calc_attractive_potential(self, pos, goal):
+        #print("is norm thing correct" + str(np.linalg.norm(np.array(pos) - np.array(goal))))
+        return 0.5 * self.KP * np.linalg.norm(np.array(pos) - np.array(goal))
 
 
-    def calc_repulsive_potential(self,x, y, ox, oy, rr):
+    def calc_repulsive_potential(self,pos, obs, rr):
```
Comment (Nabushika) [inline src/algorithms/classic/graph_based/potential_field.py:144]: whitespace for arguments

File: src/algorithms/classic/sample_based/rrt.py [hunk 3]
```diff
@@ -33,8 +33,8 @@ def _get_new_vertex(self, q_near: Vertex, q_sample: Point, max_dist) -> Vertex:
 
     def _get_random_sample(self) -> Point:
         while True:
-            sample: Point = Point(torch.randint(0, self._get_grid().size.width, (1,)).item(),
-                                  torch.randint(0, self._get_grid().size.height, (1,)).item())
+            rand_pos = [torch.randint(0, self._get_grid().size[i], (1,)).item() for i in range(self._get_grid().size.n_dim)]  
+            sample: Point = Point(*tuple(rand_pos))
```
Comment (Nabushika) [inline src/algorithms/classic/sample_based/rrt.py:37]: No need for tuple here, it's already a list we can unpack
also, can we use np? We could have `rand_pos = np.random.randint(0, self.grid.size, self.grid.size.n_dim)`?

File: tests/test_algorithm/test_configuration/test_maps/test_dense_map_3d.py [hunk 4]
```diff
@@ -166,7 +166,9 @@ def test_eq_sparse_map(self) -> None:
         self.assertEqual(map1, map2)
 
     def test_convert_to_sparse_map(self) -> None:
-        map1: DenseMap = Maps.grid_map_3d_example
+        map1: DenseMap = Maps.grid_map_3d_example_4_4_4
+        #TODO: check why this doesn't work for 3d sparse maps that have been converted to dense maps
```
Comment (Nabushika) [inline tests/test_algorithm/test_configuration/test_maps/test_dense_map_3d.py:170]: What doesn't work, and should we leave this TODO in?

**Review by Nabushika — APPROVED** [review]

**Review by Nabushika — APPROVED** [review]

**Review by judicaelclair — APPROVED** [review]

**Pull request merged**